app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt

```diff
@@ -69,42 +72,38 @@ class AnnouncementAdapter(
         }
 
         item.reactions.forEachIndexed { i, reaction ->
-            chips.getChildAt(i)?.takeUnless { it.id == R.id.addReactionChip } as Chip?
-                ?: Chip(ContextThemeWrapper(chips.context, R.style.Widget_MaterialComponents_Chip_Choice)).apply {
-                    isCheckable = true
-                    checkedIcon = null
-                    chips.addView(this, i)
-                }
-                    .apply {
-                        val emojiText = if (reaction.url == null) {
-                            reaction.name
+            (
+                chips.getChildAt(i)?.takeUnless { it.id == R.id.addReactionChip } as Chip?
+                    ?: Chip(ContextThemeWrapper(chips.context, R.style.Widget_MaterialComponents_Chip_Choice)).apply {
+                        isCheckable = true
+                        checkedIcon = null
+                        chips.addView(this, i)
+                    }
+                )
+                .apply {
+                    if (reaction.url == null) {
+                        this.text = "${reaction.name} ${reaction.count}"
+                    } else {
+                        val spanBuilder = SpannableStringBuilder("E ${reaction.count}")
```

Comment:
I got confused by this `E` for quite a bit but then I realized it's for emoji. Why can't we put the emoji itself there? Do we know the name or shortcode? This might be better for the screen reader

Reply:
Yes but putting the name there is what causes the sizing bug :/
Spans have a content description, maybe that's where the name should go? No idea what a screenreader does with it though.

Reply:
I think putting it on a span wouldn't hurt. Putting something that is usually not pronounced (like fixed-width space) instead of E could be good as well. Also a comment why we do any of this, otherwise next clueless person who touches this will break it.